src/__tests__/documenter.test.ts

```diff
@@ -2,15 +2,16 @@
 // SPDX-License-Identifier: Apache-2.0
 import { expect, test } from "vitest";
 
-import { getAllComponents, requireComponentDefinition } from "./utils";
+import componentDefinitions from "../../lib/components/internal/api-docs/components";
+import { getAllComponents } from "./utils";
 
 test.each<string>(getAllComponents())(`definition for %s matches the snapshot`, (componentName: string) => {
-  const definition = requireComponentDefinition(componentName);
+  const definition = componentDefinitions[componentName];
 
   // overriding with a fake value so that when there are icon changes in components this test doesn't block it
-  const iconNameDefinition = definition.properties.filter(({ name }: { name: string }) => name === "iconName");
-  if (iconNameDefinition && iconNameDefinition[0]) {
-    iconNameDefinition[0].inlineType.values = "comes from @cloudscape-design/components";
+  const iconNameDefinition = definition.properties.find(({ name }: { name: string }) => name === "iconName");
+  if (iconNameDefinition && iconNameDefinition.inlineType?.type === "union") {
```

Comment:
Small changes here to pass type checks (it worked before because everything was `any` type)